pkg/apiserver/resthandler.go

```diff
@@ -676,6 +676,13 @@ func UpdateResource(r rest.Updater, scope RequestScope, typer runtime.ObjectType
 			}
 		}
 
+		canDelete := func() bool {
```

Comment:
@liggitt @lavalamp done right, this would let us do:

```
# in resthandler
# patch:
storage.Update(ctx, func(ctx, old) () {
  original := Encode(old)
  updated := patch.Apply(original, newPatch)
  new := Decode(updated)
  admit.Admit(new, old)
  return new, nil
})

# preconditions
storage.Update(ctx, func(ctx, old) () {
  if !preconditions.Check(old) {
    return nil, preconditionFailure
  }
  admit.Admit(new, old)
  return new, nil
})
```

etc.